.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,26 @@
+name: golangci-lint
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - main
```

Comment:
I this a required section? Can this be just "any" for short?

Reply:
What's the benefit?

Reply:
Shortness, and fewer assumptions about how branches are named.

Reply:
I've removed all the filters. Any push to any branch, off fires the linter.

I don't see the benefit, but we can always change it.